fairscale/nn/pipe/async_schedule.py

```diff
@@ -11,15 +11,64 @@
 
 import torch
 from torch import Tensor, nn
+from torch.autograd.profiler import record_function
```

Comment:
why do we need this? Is it to all the user to gather more statistics when they do end up profiling?

Reply:
It adds tracing which can be viewed via InsightSystems or autogrid profiler. Is super handy for debugging performance and timing issues.